nemo/core/classes/mixins/access_mixins.py

```diff
@@ -48,12 +48,16 @@ def register_accessible_tensor(self, tensor):
             tensor = tensor.detach()
 
         if not hasattr(self, '_registry'):
-            self._registry = []
+            self._registry = {}
 
-        if len(self._registry) > 0:
-            self._registry.clear()
+        # incorrect logic
+        # if len(self._registry) > 0:
+        #     self._registry.clear()
```

Comment:
The reason this was here before was to avoid a situation where we are instantiating an ssl model with access enabled from a nemo file in order to then fine-tune a model which doesn't use this mixin, in which case we'll end up adding modules to registry but never resetting and run out of memory. Since reset is now present in ctc and rnnt models it won't be an issue for them, but could be for other classes.

Reply:
Hmm we can use the internal flag to check if we're being restored, and if so, avoiding activating the registry. Lemme see.